projects/addon-editor/components/editor-new/editor-new.component.ts

```diff
@@ -137,7 +149,35 @@ export class TuiEditorNewComponent
         return '';
     }
 
-    private readonly isSelectionLink = (): boolean => !!this.editor?.isActive('link');
+    private currentFocusedNodeIsAnchor(): boolean {
+        return !!this.selection?.anchorNode?.parentElement?.closest('a');
```

Comment:
`DropdownSelection` uses `TuiBooleanHandler<Range>` which means the function receives currently selected `Range` as input. You can probably use it to determine all those things and not rely on getting `Selection` manually everytime.

Reply:
unfortunately, each time the handler shows only the old value, although the cursor is already in a different place

<img width="960" alt="image" src="https://user-images.githubusercontent.com/12021443/173874440-d2ba3d56-5fb0-4ef0-9a8f-848e10df085c.png">
<img width="674" alt="image" src="https://user-images.githubusercontent.com/12021443/173874626-e8d3a626-5199-4f09-b3b7-5626ec9b6213.png">

Reply:
@waterplea can you help me with this issue?)